### PR TITLE
rte: use configfile instead of cli args

### PR DESCRIPTION
**Pull request by Tal-or**
As part of bumping RTE -> v0.20.3

We transition RTE to consume its configuration from a configfile (essentially a mounted ConfigMap) instead of commandline arguments.

This change requires adjustments all across the operator code. 
With this PR the operator creates a `ConfigMap` that contains RTE configuration and mounts it into RTE container.
After the bump to v0.19.0+ RTE knows to give precedence to configuration consumed from configfile over a command line args.

### CODE REVIEW

**Review by ffromani — COMMENTED** [review]
Nice start.

We have some e2e test which want to consume the RTE config, and/or test RTE command line flags. They likely need to be updated as well:

https://github.com/openshift-kni/numaresources-operator/pull/1075/commits/fa14b5c9cca282ce8b01ffa884f0973c40834a30#diff-49c77cd6ef9fc89514a0030c40a9073e66ef66c94ef0c3ea6a834cecda8ae6ae

**Comment by Tal-or** [comment]
We need to fix some security vulnerabilities u/s 
```
Container test exited with code 1, reason Error
---
penshift-kni/numaresources-operator ...
 ✗ [Medium] Path Traversal
   ID: 2fd9a4d3-c857-44e2-8a4b-1f487867022d 
   Path: vendor/github.com/k8stopologyawareschedwg/resource-topology-exporter/pkg/config/cfgfile.go, line 107 
   Info: Unsanitized input from file name flows into os.ReadFile, where it is used as a path. This may result in a Path Traversal vulnerability and allow an attacker to read arbitrary files.
 ✗ [Medium] Path Traversal
   ID: fdd4f27d-afea-4771-b0e8-919f66ff130e 
   Path: vendor/github.com/k8stopologyawareschedwg/resource-topology-exporter/pkg/config/cfgfile.go, line 122 
   Info: Unsanitized input from file name flows into os.ReadFile, where it is used as a path. This may result in a Path Traversal vulnerability and allow an attacker to read arbitrary files.
 ✗ [Medium] Path Traversal
   ID: 6b1a4[96](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift-kni_numaresources-operator/1090/pull-ci-openshift-kni-numaresources-operator-main-security/1861387618982301696#1:build-log.txt%3A96)9-48f1-4ac7-b85e-698fb38437e5 
   Path: vendor/github.com/k8stopologyawareschedwg/resource-topology-exporter/pkg/config/config.go, line 116 
   Info: Unsanitized input from a CLI argument flows into os.ReadDir, where it is used as a path. This may result in a Path Traversal vulnerability and allow an attacker to get a list of arbitrary files.
```

**Comment by Tal-or** [comment]
@ffromani 
Security lane is optionally. we don't need to skip

File: go.mod [hunk 1]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/k8stopologyawareschedwg/deployer v0.21.0
 	github.com/k8stopologyawareschedwg/noderesourcetopology-api v0.1.2
 	github.com/k8stopologyawareschedwg/podfingerprint v0.2.2
-	github.com/k8stopologyawareschedwg/resource-topology-exporter v0.16.1
+	github.com/k8stopologyawareschedwg/resource-topology-exporter v0.19.0
```
Comment (ffromani) [inline go.mod:16]: please let's consume v0.20.3. Changes between v0.19.0 and v0.20.3 should not create any major disruption: https://github.com/k8stopologyawareschedwg/resource-topology-exporter/compare/v0.19.0...v0.20.3

**Review by ffromani — COMMENTED** [review]
I'm not happy with this PR. It's too big of a jump. We need to consume v0.20.3 but keep using the command line args as vehicle to configure RTE. Only later we will switch to configfile.

**Comment by Tal-or** [comment]
> I'm not happy with this PR. It's too big of a jump. We need to consume v0.20.3 but keep using the command line args as vehicle to configure RTE. Only later we will switch to configfile.

Too bad we spent so much time working on this one. It would save a lot if would keep the cli args in the first place.
After all this was the original reason for closing #988

**Comment by ffromani** [comment]
> > I'm not happy with this PR. It's too big of a jump. We need to consume v0.20.3 but keep using the command line args as vehicle to configure RTE. Only later we will switch to configfile.

The fact we need to narrow down is unfortunate indeed, but we run out of time and the changes are too invasive, and more is likely needed. Reducing the scope is safer and the only option left at this point in time. We will reuse most if not all the code you posted in later updates which we can do incrementally
 
> Too bad we spent so much time working on this one. It would save a lot if would keep the cli args in the first place. After all this was the original reason for closing #988

I guess we could have done a takeover of #988, but the original reason to close it was reprioritization and rebalancing rather than anything else

**Comment by ffromani** [comment]
let us reduce further waste: https://github.com/openshift-kni/numaresources-operator/pull/1095

**Comment by Tal-or** [comment]
closed due to https://github.com/openshift-kni/numaresources-operator/pull/1090#pullrequestreview-2472867944 
we'll add those changes later on

**Pull request closed without merging**